dnf5/context.cpp

```diff
@@ -404,7 +406,22 @@ void Context::download_and_run(libdnf5::base::Transaction & transaction) {
     if (should_store_offline) {
         base.get_config().get_tsflags_option().set(libdnf5::Option::Priority::RUNTIME, "test");
     }
-    transaction.download();
+
+    if (!transaction_store_path.empty()) {
+        constexpr const char * packages_in_trans_dir{"./packages"};
+        constexpr const char * comps_in_trans_dir{"./comps"};
+        auto & destdir_opt = base.get_config().get_destdir_option();
+        destdir_opt.set(transaction_store_path / packages_in_trans_dir);
+        std::filesystem::create_directories(transaction_store_path);
+        transaction.download();
+        std::filesystem::path comps_path = transaction_store_path / comps_in_trans_dir;
+        transaction.store_comps(comps_path);
+        libdnf5::utils::fs::File transfile(transaction_store_path / "transaction.json", "w");
+        transfile.write(transaction.serialize(packages_in_trans_dir, comps_in_trans_dir));
```

Comment:
Maybe we should prompt for user confirmation if this would overwrite an existing `transaction.json`.

Reply:
That is a very good idea, thank you.